### PR TITLE
add base-normalization to dependency checker action

### LINKED ISSUE #17897: Connector dependency reporter does not include normalization changes
## Summary
- Currently the `report-connectors-dependency` action does not include normalization changes.
- It should when there is any change under `base-normalization`.

### PR DESCRIPTION
**Pull request by erohmensing**
## What
- close https://github.com/airbytehq/airbyte/issues/17897
- check `base-normalization` as well as connectors to see if changes affected it. This should help people remember when they need to push a new version of `base-normalization` due to changes

TODO before merge:
- [ ] Remove test changes - should not require approval from normalization as not actually changing these files

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py [hunk 1]
```diff
@@ -147,6 +147,7 @@ def transform_bigquery(config: Dict[str, Any]):
             "priority": config.get("transformation_priority", "interactive"),
             "threads": 8,
             "retries": 3,
+            "test": "test"
```
Comment (erohmensing) [inline airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py:150]:
```suggestion
            "test": "test"
```

File: airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def __init__(self):
         self.target_schema = "test_normalization"
         self.container_prefix = f"test_normalization_db_{self.random_string(3)}"
         self.db_names = []
+        self.test = "test"
```
Comment (erohmensing) [inline airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py:38]:
```suggestion
        self.test = "test"
```

**Review by erohmensing — COMMENTED** [review]
Things to remove before merging (leaving to keep the action comment open for now)

**Comment by erohmensing** [comment]
@evantahler in reference to my comment [here](https://github.com/airbytehq/airbyte/issues/17897#issuecomment-1310532355), I think this actually addresses what Lerin originally wanted. 

Re: Your idea to check the normalization containers, on first glance I think it may be a bit more complicated due to them not having their own Gradle files -- happy to have a go at it separately, but think this can be merged first.

**Comment by evantahler** [comment]
Screenshot 2022-11-10 at 4 14 00 PM
Probably could use some emojis to stand out like the rest of the table above

**Review by evantahler — APPROVED** [review]
This makes sense, and is a nice improvement! 
I'd suggest waiting for the 👍 form the @airbytehq/normalization group to be sure before merging.

**Review by edgao — APPROVED** [review]
this is awesome! I didn't review the python code super deeply, but it looks reasonble? And certainly the example comment is correctly including base-normalization :shrug:

:shipit: 

but maybe feature request: fetching base-normalization version (though that'll probably be easier once it's also included in the destination_definitions.yaml)

File: tools/bin/ci_check_dependency.py [hunk 3]
```diff
@@ -32,14 +34,18 @@ def main():
...
-    all_build_gradle_files = get_connectors_gradle_files(all_connectors)
+    build_gradle_files = {}
+    for connector in all_connectors:
+        connector_path = CONNECTORS_PATH + connector + "/"
+        build_gradle_files.update(get_gradle_file_for_path(connector_path))
+    build_gradle_files.update(get_gradle_file_for_path(NORMALIZATION_PATH))
 
     # Try to find dependency in build.gradle file
-    depended_connectors = list(set(get_depended_connectors(changed_modules, all_build_gradle_files)))
+    dependent_modules = list(set(get_depended_connectors(changed_modules, build_gradle_files)))
```
Comment (alafanechere) [inline tools/bin/ci_check_dependency.py:44]:
```suggestion
    dependent_modules = list(set(get_depended_modules(changed_modules, build_gradle_files)))
```
Don't you want to rename this function, as it's now tracking connectors and normalization?

**Review by alafanechere — APPROVED** [review]
Nice!

File: tools/bin/ci_check_dependency.py (hunk 3 above)
Comment (erohmensing) [reply tools/bin/ci_check_dependency.py:44]: I do! Must have missed that mid-refactor, thanks for the catch 😄

**Pull request merged**